### PR TITLE
BOAC-2718: Adds user attribute filtering to admin page

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2718

### CODE REVIEW

**Comment by lyttam** [comment]
image

File: boac/api/util.py [hunk 1]
```diff
@@ -103,12 +103,16 @@ def authorized_users_api_feed(users, sort_by='lastName'):
         profile.update({
             'id': user.id,
             'isAdmin': user.is_admin,
+            'isAdvisor': False,
             'isBlocked': user.is_blocked,
             'canAccessCanvasData': user.can_access_canvas_data,
             'deletedAt': _isoformat(user.deleted_at),
             'departments': {},
+            'isDirector': False,
```
Comment (johncrossman) [inline boac/api/util.py]: @lyttam - In my recent work on `user_session` object and front-end router, I moved `isAdvisor` and `isDirector` _out of_ the base profile JSON. For consistency, and staying aligned with db schema, can we keep those properties on the dept mappings (below)?

File: src/components/admin/Users.vue [hunk 2]
```diff
@@ -187,25 +222,38 @@ export default {
       {key: 'campusEmail'},
       {key: 'actions', label: '', class: 'user-actions'}
     ],
-    filter: null,
-    filterFields: ['depts'],
+    filter: undefined,
+    filterPermissions: ['isAdmin', 'isAdvisor', 'isDirector', 'canAccessCanvasData'],
```
Comment (johncrossman) [inline src/components/admin/Users.vue]: @lyttam - Would you mind adding our new `isScheduler` property to the list above? :)

**Comment by johncrossman** [comment]
Looks good @lyttam  I posted a couple comments.

File: boac/api/util.py (hunk 1 above)
Comment (lyttam) [reply boac/api/util.py]: @johncrossman Certainly; this was just to provide a shortcut for determining if a user has these roles in any of their departments.

File: src/components/admin/Users.vue (hunk 2 above)
Comment (lyttam) [reply src/components/admin/Users.vue]: Yes, and I can add a checkbox to filter by that attribute also.

**Comment by lyttam** [comment]
Ready for a second look @johncrossman

**Comment by johncrossman** [comment]
Thanks @lyttam

**Pull request merged**